fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -812,18 +798,45 @@ public void subscribeToRollCalls(String laoId) {
                 error -> Log.d(TAG, "Error updating Roll Call : " + error)));
   }
 
-  private void updateCurrentObjects(LaoView laoView) throws UnknownRollCallException {
+  public void subscribeToElections(String laoId) {
+    disposables.add(
+        electionRepo
+            .getElectionsObservable(laoId)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .map(
+                ids ->
+                    ids.stream()
+                        .map(
+                            id -> {
+                              try {
+                                return electionRepo.getElectionObservable(laoId, id);
+                              } catch (UnknownElectionException e) {
+                                // Election whose ids are in that list may not be absent
+                                throw new IllegalStateException(
+                                    "Could not fetch election with id " + id);
+                              }
+                            })
+                        .collect(Collectors.toList()))
+            .flatMap(
+                subjects ->
+                    Observable.combineLatest(
+                        subjects,
+                        elections ->
+                            // Sort the election list. That way it stays somewhat consistent over
+                            // the updates
+                            Arrays.stream((Election[]) elections)
+                                .sorted(Comparator.comparing(Election::getCreation).reversed())
+                                .collect(Collectors.toList())))
+            .subscribe(
+                mElections::setValue, error -> Log.d(TAG, "Error updating Roll Call : " + error)));
+  }
+
```

Comment:
It's probably work for another PR but I think we should do away with the EventSubscribe functions that observes an `Observable` and then updates a `MutableLiveData`. Then we only need injecting the needed repos in fragments

Reply:
That's part of what I have in mind concerning the refactor of the dataflow